app/src/docs/scripts/search/utils.js

```diff
@@ -0,0 +1,141 @@
+/* eslint-disable global-require, import/no-dynamic-require */
+import docsMap from '../../docsMap'
+
+const fs = require('fs')
+const remark = require('remark')
+const remarkMdx = require('remark-mdx')
+const strip = require('remark-mdx-to-plain-text')
+const lunr = require('lunr')
+const h2p = require('html2plaintext')
+const canvasModules = require('../moduleReferences/canvasModuleHelpData.json')
+
+const baseModuleRefPath = '/docs/module-reference/'
+const searchStore = {}
+
+/**
+ * Injest mdx as plain text.
+*/
+require.extensions['.mdx'] = function readMdx(module, filename) {
+    const mdxFilename = filename
+    const mdxModule = module
+    mdxModule.exports = fs.readFileSync(mdxFilename, 'utf8')
+}
+
+/**
+ * Sanitizer function.
+*/
+const toAnchor = (text) => (
+    text.trim()
+        .toLowerCase()
+        .replace(' & ', ' and ')
+        .replace(/\s+/g, '-')
+)
+
+/**
+ * Generates formatted Module text content.
+*/
+const generateModuleRefTextContent = (name, path, content) => (
+    `${name.charAt(0).toUpperCase() + name.slice(1)} (${path.charAt(0).toUpperCase() + path.slice(1)}). ${content}`
```

Comment:
Let's maybe define a `titleize` utility, pass a string to it and make it return one with first letter capitalized.